sortinghat/core/api.py

```diff
@@ -771,59 +813,60 @@ def _merge_enrollments(trxl, from_uid, to_uid):
-        if not to_uid.profile.name:
-            to_uid.profile.name = from_uid.profile.name
-        if not to_uid.profile.email:
-            to_uid.profile.email = from_uid.profile.email
-        if not to_uid.profile.gender:
-            to_uid.profile.gender = from_uid.profile.gender
-        if not to_uid.profile.gender_acc:
-            to_uid.profile.gender_acc = from_uid.profile.gender_acc
-        if not to_uid.profile.country:
-            to_uid.profile.country = from_uid.profile.country
+    def _merge_profiles(from_uids, to_uid):
+        """Merge the profiles from `UniqueIdentity` objects"""
+
+        for from_uid in from_uids:
+            if from_uid.profile.is_bot or to_uid.profile.is_bot:
+                to_uid.profile.is_bot = True
+            if not to_uid.profile.name:
+                to_uid.profile.name = from_uid.profile.name
+            if not to_uid.profile.email:
+                to_uid.profile.email = from_uid.profile.email
+            if not to_uid.profile.gender:
+                to_uid.profile.gender = from_uid.profile.gender
+            if not to_uid.profile.gender_acc:
+                to_uid.profile.gender_acc = from_uid.profile.gender_acc
+            if not to_uid.profile.country:
+                to_uid.profile.country = from_uid.profile.country
 
         return to_uid
 
+    def _delete_unique_identities(trxl, uids):
+        """Delete unique identities from the database"""
+
+        for uid in uids:
+            delete_unique_identity_db(trxl, uid)
+
     # Check input values
-    if from_uuid is None:
-        raise InvalidValueError(msg="'from_uuid' cannot be None")
-    if from_uuid == '':
-        raise InvalidValueError(msg="'from_uuid' cannot be an empty string")
+    if from_uuids is None:
+        raise InvalidValueError(msg="'from_uuids' cannot be None")
+    if from_uuids == []:
+        raise InvalidValueError(msg="'from_uuids' cannot be an empty list")
     if to_uuid is None:
         raise InvalidValueError(msg="'to_uuid' cannot be None")
     if to_uuid == '':
         raise InvalidValueError(msg="'to_uuid' cannot be an empty string")
-    if from_uuid == to_uuid:
-        raise InvalidValueError(msg="'from_uuid' and 'to_uuid' cannot be equal")
 
     trxl = TransactionsLog.open(name='merge_identities')
 
     try:
-        from_uid = find_unique_identity(from_uuid)
         to_uid = find_unique_identity(to_uuid)
+        from_uids = _find_unique_identities(from_uuids, to_uuid)
     except NotFoundError as exc:
-        # Identities were not found, so they cannot be merged
+        # At least one unique identity was not found, so they cannot be merged
         raise exc
 
-    try:
-        identities = from_uid.identities.all()
-    except ValueError as e:
-        raise InvalidValueError(msg=str(e))
-
-    for identity in identities:
-        move_identity_db(trxl, identity, to_uid)
+    identities = _get_identities(from_uids)
+    _move_identities(trxl, identities, to_uid)
```

Comment:
Maybe it's better to call `_get_identities` inside `_move_identities` because `identities` variable is not used anywhere else. It also will match with the prototype of the other functions.